File: test7C#/test7/Model/Drink/LongIslandIceTea.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace test7.Model.Drink
{
    class LongIslandIceTea : SimpleDrink
    {
        public LongIslandIceTea() : base("Long Island Ice Tea", Convert.ToDecimal(19.47))
        {
            base.Composition(new List<Dosage>() { 
                new Dosage("Ice Tea", 3),
                new Dosage("Vodka", 1),
                new Dosage("Rhum", 0.5f),
                new Dosage("Pineapple juice", 0.5f),
                new Dosage("Peach juice", 0.5f),
                new Dosage("Cayen", 0.01f)
            });;
        }
    }
}

File: test7Go/oop/models/drink/LongIslandIceTea.go
package drink

type LongIslandIceTea struct {
	SimpleDrink
}

func NewLongIslandIceTea() LongIslandIceTea {
	simpleDrink := NewSimpleDrink("Long Island Ice Tea", 19.47)
	simpleDrink.Incrediants = "Ice Tea, Vodka, Rhum, Pineapple Juice, Peach Juice, Cayen"

	return LongIslandIceTea{simpleDrink}
}
